lib/Models/CartoMapCatalogItem.ts

```diff
@@ -0,0 +1,143 @@
+import CreateModel from "./CreateModel";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import CartoMapCatalogItemTraits from "../Traits/CartoMapCatalogItemTraits";
+import Mappable from "./Mappable";
+import proxyCatalogItemUrl from "./proxyCatalogItemUrl";
+import { computed, runInAction } from "mobx";
+import TerriaError from "../Core/TerriaError";
+import UrlTemplateImageryProvider from "terriajs-cesium/Source/Scene/UrlTemplateImageryProvider";
+import Resource from "terriajs-cesium/Source/Core/Resource";
+import Rectangle from "terriajs-cesium/Source/Core/Rectangle";
+import isDefined from "../Core/isDefined";
+import CommonStrata from "./CommonStrata";
+
+export default class CartoMapCatalogItem
+  extends UrlMixin(CatalogMemberMixin(CreateModel(CartoMapCatalogItemTraits)))
+  implements Mappable {
+  static readonly type = "carto";
+
+  get type() {
+    return CartoMapCatalogItem.type;
+  }
+
+  get isMappable() {
+    return true;
+  }
+
+  loadMapItems(): Promise<void> {
+    if (isDefined(this.tileUrl)) {
+      return Promise.resolve();
+    }
+
+    let queryParameters: { auth_token?: string };
+    queryParameters = {};
+    if (this.auth_token) {
+      queryParameters.auth_token = this.auth_token;
+    }
+
+    if (this.url === undefined) {
+      throw new TerriaError({
+        title: "Unable to load Carto Map layer",
+        message:
+          "The Carto Map layer cannot be loaded the catalog item does not have a `url`."
+      });
+    }
+
+    const resource = new Resource({
+      url: this.url,
+      headers: {
+        "Content-Type": "application/json"
+      },
+      queryParameters: queryParameters
+    });
+
+    let promise = resource.post(JSON.stringify(this.config || {}));
+    if (promise === undefined) {
+      throw new TerriaError({
+        sender: this,
+        title: "Could not load Carto Map layer",
+        message: "There was an error loading the data for this catalog item."
+      });
+    }
+
+    return promise.then(response => {
+      const map = JSON.parse(response);
+
+      let url: string;
+      let subdomains: string[];
+
+      const metadataUrl =
+        map.metadata && map.metadata.url && map.metadata.url.raster;
+      if (metadataUrl) {
+        url = metadataUrl.urlTemplate;
+        subdomains = metadataUrl.subdomains;
+      } else {
+        throw new TerriaError({
+          sender: this,
+          title: "No URL",
+          message: "Unable to find a usable URL for the Carto Map layer."
+        });
+      }
+
+      runInAction(() => {
+        this.setTrait(CommonStrata.user, "tileUrl", url);
```

Comment:
A better way to load this extra metadata is to create a LoadableStratum like in [WMS catalog item](https://github.com/TerriaJS/terriajs/blob/mobx/lib/Models/WebMapServiceCatalogItem.ts), but instead of calling the static `.load` function in `forceLoadMetadata` put the load in `forceLoadMapData` and add the `AsyncMappableMixin` to your class

This ensures that if traits are changed the visualisation will be kept up to date.